### PR TITLE
fix: updating target auth implementation

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
* Updating target_auth implementation to leverage property in the target_account.js file to avoid more template files
* Updating upstream components to leverage new approach
* Updating README guidance on using the component

Will follow up with a PR against the guidance repo to leverage new integration.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/osml/osml_auth_server.ts [hunk 1]
```diff
@@ -76,17 +45,17 @@ export interface OSMLAuthenticateProps {
 /**
  * Represents the construct that authenticates the user.
  */
-export class OSMLAuthenticate extends Construct {
+export class OSMLAuthServer extends Construct {
```
Comment (jtblack-aws) [inline lib/osml/osml_auth_server.ts:48]: I don't think this is the construct for the AuthServer (keycloak), is it?

Comment (drduhe) [reply lib/osml/osml_auth_server.ts:48]: What should this be called?

Comment (RanbirAulakh) [reply lib/osml/osml_auth_server.ts:48]: +1 to Jonathan. This construct is focusing on adding listener / rules to the ALB. So, we can rename something to like `OSMLAuth` or `OSMLAuthListenerRules`

File: lib/index.ts [hunk 2]
```diff
@@ -28,7 +28,8 @@ export * from "./osml/model_endpoint/me_http_endpoint";
 export * from "./osml/osml_table";
 export * from "./osml/osml_topic";
 export * from "./osml/osml_vpc";
-export * from "./osml/osml_authenticate";
+export * from "./osml/osml_auth";
+export * from "./osml/osml_auth_server";
```
Comment (jtblack-aws) [inline lib/index.ts:32]: the file should probably be renamed to osml_auth_alb to match

Comment (drduhe) [reply lib/index.ts:32]: ty ty

Comment (drduhe) [reply lib/index.ts]: fixed

**Review by jtblack-aws — APPROVED** [review]

**Pull request merged**